src/App.jsx

```diff
@@ -14,70 +17,78 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+export const App = () => {
+  const DEFAULT_PRODUCT = 'Jam';
+  const [selectedProduct, setSelectedProduct] = useState(DEFAULT_PRODUCT);
+  const goodsWithId = goods.reduce((goodsList, product, index) => {
+    goodsList.push({
+      product,
+      id: index + 1,
+    });
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return goodsList;
+  }, []);
```

Comment:
Several issues here

1. When you just 'mapping' one array onto another (i.e. each element of the resulting array is created from an element of the input array), prefer `map`
2. I would not do this at all - the goods are unique, and in your particular case there is no need to add extra data. All you are using ID for is the key - but the good name itself is perfectly fine for the key (as it is the unique string)